runtest.py

```diff
@@ -111,255 +121,188 @@ def kill_legacy_process(self):
+                            test_case_module_name = "%s.%s" % (self.global_config['default-test-dn'], "test_pilot_run")
+                            runtime_case_data["SIKULI_SCRIPT_PATH"] = test_case
+                            runtime_case_data["TEST_SCRIPT_PY_DIR_PATH"] = os.sep.join(test_case_module_name.split(".")[:-1])
+                            if test_case.endswith(os.sep):
+                                test_name = test_case.split(os.sep)[-2].split(".")[0]
+                            else:
+                                test_name = test_case.split(os.sep)[-1].split(".")[0]
+                            test_case_fp = test_case
                         else:
-                            test_name = test_case.split(os.sep)[-1].split(".")[0]
-                    else:
-                        test_case_fp = test_case.replace(".", os.sep) + ".py"
-                        test_name = test_case.split(".")[-1]
-                        case_data[test_case]["TEST_SCRIPT_PY_DIR_PATH"] = os.sep.join(test_case.split(".")[:-1])
-                        if os.path.exists(test_case_fp):
+                            test_case_fp = test_case.replace(".", os.sep) + ".py"
+                            test_name = test_case.split(".")[-1]
                             test_case_module_name = test_case
-                            test_env = self.get_test_env(**case_data[test_case])
-                        else:
-                            self.logger.error("Test script [%s] is not exist!" % test_case_fp)
-                            test_env = None
-
-                    # if webdriver is enable, we need to get parameters for running browsers
-                    if self.webdriver['enable']:
-                        test_env['webdriver'] = "True"
-                        if self.webdriver['run_firefox']:
-                            test_env['browser_type'] = "firefox"
-                            data = self.loop_test(test_case_module_name, test_name, test_env)
-                        if self.webdriver['run_chrome']:
-                            test_env['browser_type'] = "chrome"
-                            data = self.loop_test(test_case_module_name, test_name, test_env)
-                    else:
-                        data = self.loop_test(test_case_module_name, test_name, test_env)
-
-                    if self.online:
-                        if self.perfherder_revision:
-                            self.upload_agent_obj.upload_register_data(input_suite_fp, type, self.perfherder_suitename)
-                        self.upload_agent_obj.upload_videos(data)
+                            runtime_case_data["TEST_SCRIPT_PY_DIR_PATH"] = os.sep.join(test_case.split(".")[:-1])
+                            # if webdriver is enable, we need to get parameters for running browsers
+                            if self.exec_config['user-simulation-tool'] == self.global_config['default-user-simulation-tool-webdriver']:
```

Comment:
Can you put:
```
if os.path.exists(test_case_fp):
    test_env = self.get_test_env(**runtime_case_data)
else:
    self.logger.error("Test script [%s] is not exist!" % test_case_fp)
    continue
```
before this if-else for webdriver/sikuli?

It will reduce complexity inside the if-else. Also, there is no need to check the same case twice in this if-else.